nana/nana.go

```diff
@@ -0,0 +1,200 @@
+package main
+
+import (
+	"bytes"
+	"errors"
+	"flag"
+	"fmt"
+	"go/ast"
+	"go/parser"
+	"go/printer"
+	"go/token"
+	"os"
+)
+
+func main() {
+
+	var fNames string
+	patcher := flag.NewFlagSet("sew", flag.ExitOnError)
+	saveAs := patcher.String("saveAs", "patched.go", "the file to write to disk")
+	patcher.StringVar(&fNames, "files", "", "the files to sew together")
+	remove := patcher.Bool("D", false, "delete files after patching")
+
+	deleter := flag.NewFlagSet("delete", flag.ExitOnError)
+	funcName := deleter.String("func", "", "function to patch -- example: -func nameOfFunc")
+	fileName := deleter.String("file", "", "the path to the file to delete -- example: -file path/to/file")
+
+	if len(os.Args) < 3 {
+		fmt.Println("not enough arguments passed")
+		os.Exit(51)
+	}
+
+	if os.Args[1] == "delete" {
+		deleter.Parse(os.Args[2:])
+		fmt.Println("deleting ", *fileName)
+		delete(*fileName, *funcName)
+	} else if os.Args[1] == "sew" {
+		patcher.Parse(os.Args[2:])
+		args := patcher.Args()
+		fmt.Println(*saveAs)
+		fmt.Println(args)
+		patch(*saveAs, *remove, args...)
+	}
+
+	fmt.Printf("command '%s' not recognized", os.Args[1])
+	os.Exit(50)
+
+}
+
+func patch(saveAs string, deleteFiles bool, files ...string) {
+	if len(files) < 1 {
+		panic("patch requires at least 2 files to merge")
+	}
+	file, fset, err := Patch(files[0], files[1:]...)
+	if err != nil {
+		panic(err)
+	}
+	bz := getBytesFromFile(fset, file)
+	os.WriteFile(saveAs, bz.Bytes(), 0766)
+	if deleteFiles {
+		for _, f := range files {
+			os.Remove(f)
+		}
+	}
+}
+
+func delete(funcName, fileName string) {
```

Comment:
I think that delete overlaps a standard library file name, maybe a more clear name would be good.